internal/converter/paths.go

```diff
@@ -14,44 +14,182 @@ import (
 )
 
 func addPathItemsFromFile(opts options.Options, fd protoreflect.FileDescriptor, paths *v3.Paths) error {
-	services := fd.Services()
-	for i := 0; i < services.Len(); i++ {
-		service := services.Get(i)
-		methods := service.Methods()
-		for j := 0; j < methods.Len(); j++ {
-			method := methods.Get(j)
-			pathItems := googleapi.MakePathItems(opts, method)
-			for pair := pathItems.First(); pair != nil; pair = pair.Next() {
-				path, item := pair.Key(), pair.Value()
-				if existing, ok := paths.PathItems.Get(pair.Key()); !ok {
-					paths.PathItems.Set(path, item)
-				} else {
-					if item.Get != nil {
-						existing.Get = item.Get
+		services := fd.Services()
+		for i := 0; i < services.Len(); i++ {
+			service := services.Get(i)
+			methods := service.Methods()
+			for j := 0; j < methods.Len(); j++ {
+				method := methods.Get(j)
+				pathItems := googleapi.MakePathItems(opts, method)
+				
+				// Helper function to update or set path items
+				updatePathItem := func(path string, newItem *v3.PathItem) {
+					if existing, ok := paths.PathItems.Get(path); !ok {
+						paths.PathItems.Set(path, newItem)
+					} else {
+						if opts.MergeBase {
+							mergePathItems(existing, newItem)
+						} else {
+							updateOperations(existing, newItem)
+						}
+						paths.PathItems.Set(path, existing)
 					}
-					if item.Post != nil {
-						existing.Post = item.Post
-					}
-					if item.Delete != nil {
-						existing.Delete = item.Delete
-					}
-					if item.Put != nil {
-						existing.Put = item.Put
-					}
-					if item.Patch != nil {
-						existing.Patch = item.Patch
-					}
-					paths.PathItems.Set(path, existing)
+				}
+
+				// Update path items from google.api annotations
+				for pair := pathItems.First(); pair != nil; pair = pair.Next() {
+					updatePathItem(pair.Key(), pair.Value())
+				}
+
+				// Default to ConnectRPC/gRPC path if no google.api annotations
+				if pathItems == nil || pathItems.Len() == 0 {
+					path := "/" + string(service.FullName()) + "/" + string(method.Name())
+					updatePathItem(path, methodToPathItem(opts, method))
 				}
 			}
-			// No google.api annotations for this method, so default to the ConnectRPC/gRPC path
-			if pathItems == nil || pathItems.Len() == 0 {
-				paths.PathItems.Set("/"+string(service.FullName())+"/"+string(method.Name()), methodToPathItem(opts, method))
+		}
+
+	return nil
+}
+
+// Helper function to update operations
+func updateOperations(existing, newItem *v3.PathItem) {
+	operations := []struct {
+		existingOp **v3.Operation
+		newOp      *v3.Operation
+	}{
+		{&existing.Get, newItem.Get},
+		{&existing.Post, newItem.Post},
+		{&existing.Delete, newItem.Delete},
+		{&existing.Put, newItem.Put},
+		{&existing.Patch, newItem.Patch},
+	}
+
+	for _, op := range operations {
+		if op.newOp != nil {
+			*op.existingOp = op.newOp
+		}
+	}
+}
```

Comment:
I like the simplicity of this. Can this same style be applied to `mergePathItems`? Also, this method can probably be removed if e default to always merging